base/sparse/sparsematrix.jl

```diff
@@ -431,6 +431,37 @@ end
 
 ## Unary arithmetic and boolean operators
 
+macro _unary_op(Ti,Tv,A,op)
```

Comment:
I like the idea of starting to reduce repetition with these kind of template macros, but we should try to come up with more descriptive names for them.